File: python/questions/myPow.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        pn = n
        if n < 0:
            x = 1 / x
            pn = -pn
        ans = 1
        while pn > 0:
            ans *= x if pn&1 else 1
            x *= x
            pn >>= 1
        return ans

File: cpp/questions/myPow.cpp
class Solution {
public:
    double myPow(double x, int n) {
        long long pn = n;
        if (n < 0) {
            x = 1 / x;
            pn = -pn;
        }
        double ans = 1;
        while (pn > 0) {
            bool bv = pn & 1;
            ans *= bv ? x : 1;
            x *= x;
            pn >>= 1;
        }
        return ans;
    }
};
